Example added.

amitjamadagni opened the pull request:
Quantum Harmonic Oscillator, usage of QuDyamic solvers.

amitjamadagni commented:
@acroy is this ready for the merge ?

acroy commented on examples/jaynes_cummings_mcwf.jl:
```diff
@@ -0,0 +1,59 @@
+
+# Calculating the expectation of a'*a and sm'*sm at every time step
+for i in 1:length(tlist)-1
+    trace(rho[i]*a'* a)
```
You should either print those expectation values or store them in some array.

acroy commented on examples/jaynes_cummings.jl:
```diff
@@ -0,0 +1,58 @@
+
+c_ops = result[2]
+
+qprop = QuPropagator(H, c_ops, psi0*psi0', tlist, QuODE45())
```
I guess there is something missing after that line?

amitjamadagni commented:
@acroy the I have added the printing of `trace`.

acroy commented on examples/jaynes_cummings.jl:
```diff
@@ -0,0 +1,65 @@
+a = tensor(lowerop(2), QuArray(eye(2)))
+op = a'*a
+for (t, rho) in qprop
+    println(trace(rho*op))
```
Maybe this could be a bit more verbose :-) Like `println("t = $t : <a'*a> = $(trace(rho*op))")`.

acroy commented on examples/jaynes_cummings.jl:
```diff
@@ -0,0 +1,65 @@
+using QuBase
+using QuDynamics
+
+function jc_integrate(N, wc, wa, g, kappa, gamma, psi0, use_rwa, tlist)
```
I guess this name comes from the qutip example, but here it doesn't make much sense. Maybe `jc_setup` would be better?

acroy commented on examples/jaynes_cummings.jl:
```diff
@@ -0,0 +1,65 @@
+
+tlist = 0.:0.1:2*pi
+
+result = jc_integrate(N, wc, wa, g, kappa, gamma, psi0, use_rwa, tlist)
```
Just use `H, c_ops = jc_integrate(N, wc, wa, g, kappa, gamma, psi0, use_rwa, tlist)`.

acroy commented on examples/jaynes_cummings.jl:
```diff
@@ -0,0 +1,65 @@
+# We thank QuTiP developers for sharing the notebook on Jaynes Cummings model,
```
Might be good to give a link to the original notebook.

acroy commented on examples/jaynes_cummings_mcwf.jl:
```diff
@@ -0,0 +1,59 @@
+
+# Calculating the expectation of the observable a'*a and sm'*sm at every time step
+for i in 1:length(tlist)-1
+    println(trace(rho[i]*a'* a))
```
Same as above. Maybe `println("t = $(tlist[i+1]) : <a'*a> = $(trace(rho[i]*a'*a)), <sm'*sm> = $(trace(rho[i]*sm'* sm))")` would be better.

acroy commented on examples/README.md:
```diff
@@ -0,0 +1,23 @@
+
+1. Implementation of Jaynes Cummings Model
+
+[QuDynamics version](https://github.com/amitjamadagni/Notes-Notebooks/blob/master/Notebooks/JC%20model%20-%20%20QuTiP%20in%20QuDynamics.ipynb)
```
Note that those links point to your (private) repo. I guess we want to have those notebooks in the QuDynamics repo?

amitjamadagni replied:
May be we could have it in the resources, the notebooks might populate the repo ?

amitjamadagni replied:
@acroy added the notebooks section. The links should be working once we have this merged.

acroy replied:
You forgot to change the first two links. Could you also change the file name "JC_model ..." to "JC_Model ..." Or vice versa?

amitjamadagni commented:
@acroy I have changed the links (the link of QuTiP was a tweaked version of original implementation which involved comparing of results). I have replaced it though. I guess this is ready !

acroy commented:
Should be fine.

amitjamadagni commented:
Merging this !
